tests/integrationtests/tst_perfparser.cpp

```diff
@@ -181,6 +181,90 @@ private slots:
         m_machineHostName = QSysInfo::machineHostName();
     }
 
+    void testFileErrorHandling()
+    {
+        PerfParser parser(this);
+        QSignalSpy parsingFailedSpy(&parser, &PerfParser::parsingFailed);
+        QString message;
+
+        const auto notThereFile = QStringLiteral("not_here");
+        parser.initParserArgs(notThereFile);
+
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(notThereFile));
+        QVERIFY(message.contains(QLatin1String("does not exist")));
+
+        const auto parentDirs = QStringLiteral("../..");
+        parser.initParserArgs(parentDirs);
+        // note: initializing parser args reset the attached spy counter
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(parentDirs));
+        QVERIFY(message.contains(QLatin1String("is not a file")));
+
+        const auto noRead = QStringLiteral("no_r_possible");
+        const auto shell = QStandardPaths::findExecutable(QStringLiteral("bash"));
+        if (!shell.isEmpty()) {
+            QProcess::execute(
+                shell, {QLatin1String("-c"), QStringLiteral("rm -rf %1 && touch %1 && chmod a-r %1").arg(noRead)});
+            parser.initParserArgs(noRead);
+
+            QCOMPARE(parsingFailedSpy.count(), 1);
+            message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+            QVERIFY(message.contains(noRead));
+            QVERIFY(message.contains(QLatin1String("not readable")));
+        }
+    }
+
+    void testFileContent_data()
+    {
+        QTest::addColumn<QString>("perfFile");
+        QTest::addColumn<QString>("errorMessagePart");
+        QTest::addColumn<int>("waitTime");
+
+        QTest::addRow("pre-exported perfparser") << QFINDTESTDATA("file_content/true.perfparser") << QString() << 2000;
+        QTest::addRow("invalid data") << QFINDTESTDATA("tst_perfparser.cpp") << QStringLiteral("invalid perf data file")
+                                      << 1000;
+        QTest::addRow("PERF v1") << QFINDTESTDATA("file_content/perf.data.true.v1")
+                                 << QStringLiteral("invalid perf data file") << 1000;
+
+        /* FIXME: this waittime and the one below must be so high as unavailable elf files are checked
+           again and again via PerfSymbolTable::registerElf, as seen in the output */
+        QTest::addRow("PERF v2") << QFINDTESTDATA("file_content/perf.data.true.v2") << QString() << 8000;
+#if KFArchive_FOUND
+        QTest::addRow("PERF v2 gz") << QFINDTESTDATA("file_content/perf.data.true.v2.gz") << QString() << 9000;
+#endif
+    }
+
+    void testFileContent()
+    {
+        Settings::instance()->setAppPath(
```

Comment:
why does this matter, can you please add a comment? we shouldn't need to do this

Reply:
The AppPath? That removes noise because the binary for the recording is not necessarily in the same place as the one recorded. See #551 for the problem, should I reference this issue here and let it be?
If/when this issue is solved then _maybe_ this isn't needed any more, but this PR actually gives you the direct option to test and possibly do something about it.